src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -369,4 +346,35 @@ public ApplianceBootstrapInformationElement getBootstrapinfo(BootStrapInfoProvid
             }
         };
     }
+
+    private DeviceMemberEntity getDeviceMember(Long deviceId, Long memberId, String memName) {
+        CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+        CriteriaQuery<DeviceMemberEntity> query = criteriaBuilder.createQuery(DeviceMemberEntity.class);
+        Root<DeviceMemberEntity> r = query.from(DeviceMemberEntity.class);
+
+        query.select(r).where(criteriaBuilder.and((criteriaBuilder.equal(r.get("device").get("id"), deviceId)),
+                criteriaBuilder.equal(r.get("id"), memberId)));
+
+        if (memName != null) {
+            query.select(r)
+            .where(criteriaBuilder.and((criteriaBuilder.equal(r.get("device").get("id"), deviceId)),
+                    (criteriaBuilder.equal(r.get("name"), memName))));
+        } else {
+            query.select(r).where(criteriaBuilder.and((criteriaBuilder.equal(r.get("device").get("id"), deviceId)),
```

Comment:
why is the code here and line 355 duplicated? you've already add the criteria....you are adding it again. It might result in a runtime error or double joins...

Reply:
Yes it is a mistake and cut and paste issue and will correct it.